### DISCUSSION TITLE
Class name minification

### DISCUSSION THREAD

**Discussion by privatenumber** [windicss/windicss#460] [Ideas]
If WindiCSS knows all the utility classes being used via static analysis, I'm thinking the used classes can all be obfuscated and shortened to a short name in production.

This could possible shave some bytes across the JS since long classname lists can be taxing.

**Comment by antfu** [comment]
We have a compile mode (you can play it by scrolling to the bottom of https://windicss.org/ and click the "interpret". However, a shame that it only supported by certain integrations, in this case, either vite nor webpack support this. This is mainly because it would be tricky to detect which parts are actual class names that will injected into dom in JS and do the replacement safely (or we will need to reply on certain syntax like ```let className = windi`text-red-300 bg-red-100` ``` which could still be tricky to handle and verbose to write).

Do you have any better idea in mind?

**Reply by privatenumber** [reply]
Wow that's magic! I also wanted to recommend class consolidation optimization but didn't have any ideas on how but seems like it's all there 🤯.

(I searched for "minification", "optimization", and now "compilation" and "interpret" in the docs but no results showed up. If there's a page for it, I would tag it with those keywords since this is a great feature. Seems there's some description about it on the CLI page.)

Ah that's too bad. I'm using Vite. I think it would be nice if WindiCSS offered a limited version of this feature in Vite to start with. As you say, something like enabling `interpret` flag that would only parse `class` attributes and warn on any other usage.
